src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,222 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+class Parser {
+    /**
+     * Returns a Task recovered from the record.
+     * @param record contain the information of a task
+     * @return the task recovered from the record.
+     */
+    static Task convertRecordToTask(String record) {
+        String[] strings = record.split("&&");
+        switch (strings[0]) {
+            case "T":
+                return new Todo(strings[2], strings[1].equals("Done"));
+            case "D":
+                return new Deadline(strings[2], strings[3], strings[1].equals("Done"));
+            case "E":
+                return new Event(strings[2], strings[3], strings[1].equals("Done"));
+            default:
+                return null;
+        }
+    }
+
+    static String parseInput(String input, TaskList taskList, Ui ui) {
+        String[] infos = input.split(" ", 2);
+        String type = infos[0];
+        switch (type) {
+            case "bye":
+                ui.showGoodbye();
+                System.exit(0);
+                //return null;
+            case "list":
+                if (infos.length == 1) {
+                    ui.printList(taskList);
+                    return null;
+                } else {
+                    return "Wrong input format";
+                }
+            case "schedule":
+                return getSchedule(input, taskList);
+            case "done":
+                if (infos.length == 2) {
+                    String index = infos[1];
+                    return doTask(index, taskList);
+                } else {
+                    return "Wrong input format";
+                }
+            case "find":
+                if (infos.length == 2) {
+                    String content = infos[1];
+                    return findContent(content, taskList);
+                } else {
+                    return "Wrong input format";
+                }
+            case "delete":
+                if (infos.length == 2) {
+                    String index = infos[1];
+                    return deleteTask(index, taskList);
+                } else {
+                    return "Wrong input format";
+                }
+            default:
+                return addList(input, taskList);
+        }
+    }
+
+
+    /**
+     * Returns error messages or a string showing the added task.
+     * The input string must follow input format, otherwise error messages will be return.
+     * The new task will be created and added to lst.
+     * @param input input message
+     * @param taskList the taskList containing all tasks
+     * @return a string showing the added task and number of tasks
+     */
+     static String addList(String input, TaskList taskList) {
+         ArrayList<Task> lst = taskList.getTasks();
+         ArrayList<String> validType = new ArrayList<>(
```

Comment:
```suggestion
         ArrayList<Task> tasks = taskList.getTasks();
         ArrayList<String> validTypes = new ArrayList<>(
```
Would it be better to name your variables in plural form instead, to show that they are collections of items? See [this](https://nus-cs2103-ay2122s1.github.io/website/book/codeQuality/nameWell/basic/nounsAndVerbsAsNames/) section of the textbook.